backend/src/v2/metadata/client.go

```diff
@@ -576,6 +582,24 @@ func (c *Client) CreateExecution(ctx context.Context, pipeline *Pipeline, config
 			},
 		}}
 	}
+	// We save the output parameter and output artifact relationships in MLMD in
+	// case they're provided by a sub-task so that we can follow the
+	// relationships and retrieve outputs downstream in components that depend
+	// on said outputs as inputs.
+	if config.OutputParameters != nil {
+		e.CustomProperties[keyOutputs] = &pb.Value{Value: &pb.Value_StructValue{
+			StructValue: &structpb.Struct{
+				Fields: config.OutputParameters,
+			},
+		}}
+	}
+	if config.OutputArtifacts != nil {
+		b, err := json.Marshal(config.OutputArtifacts)
+		if err != nil {
+			return nil, err
+		}
+		e.CustomProperties[keyOutputArtifacts] = StringValue(string(b))
+	}
```

Comment:
one thing I'm thinking about is how we distinguish between `outputs` for container executions vs DAG executions, since for paramers in container executions they map to the actual resolved values, but for DAG we're simply storing reference values to the producer tasks

similarly for artifacts, for container executions we store the artifact obj store metadata (pulled from artifact properties), but for dag executions we are again, storing the producer metadata, right? 

so I'm wondering if it makes sense to have a separate parameter entirely to distinguish these types, I don't have a concrete suggestion, maybe something like parameter_producer_tasks and artifact_producer_tasks 🤷🏾‍♂️ 


I'm not sure but atm if we use `outputs` then it will probably show up on the UI for executions page under the DAG but show something different than it does for Container executions